Python/bytecodes.c

```diff
@@ -2955,18 +2954,35 @@ dummy_func(
             PyCodeObject *code = (PyCodeObject *)func->func_code;
             DEOPT_IF(code->co_argcount != argcount, CALL);
             DEOPT_IF(!_PyThreadState_HasStackSpace(tstate, code->co_framesize), CALL);
+        }
+
+        op(_INIT_CALL_PY_EXACT_ARGS, (method, callable, args[oparg] -- new_frame: _PyInterpreterFrame*)) {
+            int is_meth = method != NULL;
+            int argcount = oparg;
+            if (is_meth) {
+                callable = method;
+                args--;
+                argcount++;
+            }
```

Comment:
It's a shame that this logic needs to be duplicated.

Also, not sure if you saw #107788, but one of us is going to have conflicts. It's not too bad, though.

Reply:
It's a bloody shame, but I don't know how to avoid it. We can put effort into this later if we feel it's needed.

Do you expect to be landing your PR before you head out on vacation? In that case I think you should go first. I expect that my stuff gets slightly simpler after that. I'll review it in a bit.

Reply:
Yeah, I'd like to land mine by tomorrow if possible. Thanks.